mods/ra/maps/fall-of-greece-1-personal-war/personal-war.lua

```diff
@@ -0,0 +1,492 @@
+	if not ChurchDamaged then
+		Reinforcements.Reinforce(greece, CivilianSquad, { ChurchSpawn.Location, GrenEntry.Location }, 0)
+	end
+	ChurchDamaged = true
+end
+
+SendMig = function(waypoints)
+	local MigEntryPath = { waypoints[1].Location, waypoints[2].Location }
+	local Mig = Reinforcements.Reinforce(ussr, { "mig" }, MigEntryPath)
+	if not BridgeBarrel1.IsDead then
+		Utils.Do(Mig, function(mig)
+			mig.Attack(BridgeBarrel1)
+		end)
+	end
+end
+
+SetupTriggers = function()
+	Trigger.OnKilled(BridgeBarrel1, function()
+		local bridge = Utils.Where(Map.ActorsInWorld, function(actor) return actor.Type == "bridge2" end)[1]
+		if not bridge.IsDead then
+			bridge.Kill()
+		end
+	end)
+
+	Trigger.OnKilled(BridgeBarrel2, function()
+		local bridgepart1 = Utils.Where(Map.ActorsInWorld, function(actor) return actor.Type == "br2" end)[1]
+		local bridgepart2 = Utils.Where(Map.ActorsInWorld, function(actor) return actor.Type == "br3" end)[1]
+		if not bridgepart1.IsDead then
+			bridgepart1.Kill()
+		end
+		if not bridgepart2.IsDead then
+			bridgepart2.Kill()
+		end
+	end)
+
+	local trukEscaped
+	Trigger.OnEnteredFootprint({ TruckGo.Location }, function(actor, id)
+		if actor.Type == "truk" and not trukEscaped then
+			Trigger.RemoveFootprintTrigger(id)
+			trukEscaped = true
+			actor.Destroy()
+		end
+	end)
+end
+
+SovBaseAttack = function()
+	if not BaseDamaged then
+		local drop3 = rifledropS.ActivateParatroopers(SovBaseDrop.CenterPosition, 191)
+		Utils.Do(drop3, function(a)
+			Trigger.OnPassengerExited(a, function(t, p)
+				IdleHunt(p)
+			end)
+		end)
+
+		Utils.Do(SovBaseTeam, function(actor)
+			if not actor.IsDead then
+				IdleHunt(actor)
+			end
+		end)
+
+		if Map.LobbyOption("difficulty") == "hard" then
+			local barracksTeam = Reinforcements.Reinforce(ussr, RaxTeam, { SovRaxSpawn.Location, SovBaseCam.Location }, 0)
+			Utils.Do(barracksTeam, IdleHunt)
+		end
+	end
+	BaseDamaged = true
+end
+
+ExtractUnits = function(extractionUnit, pos, after)
+	if extractionUnit.IsDead or not extractionUnit.HasPassengers then
+		return
+	end
+
+	extractionUnit.Move(pos)
+	extractionUnit.Destroy()
+
+	Trigger.OnRemovedFromWorld(extractionUnit, after)
+end
+
+SendExtractionHelicopter = function()
+	ExtractionHeli = Reinforcements.ReinforceWithTransport(allies, ExtractionHelicopterType, nil, ExtractionPath)[1]
+	local exitPos = CPos.New(ExtractionPath[1].X, ExtractionPath[2].Y)
+
+	Trigger.OnKilled(ExtractionHeli, function() ussr.MarkCompletedObjective(ussrObj) end)
+	Trigger.OnAllRemovedFromWorld(VIPs, function()
+		ExtractUnits(ExtractionHeli, exitPos, function()
+			allies.MarkCompletedObjective(ProtectVIPs)
+			allies.MarkCompletedObjective(ExtractStavros)
+		end)
+	end)
+end
+
+Tick = function()
+	VIPs = Utils.Where(allies.GetActors(), function(actor)
+		return
+			actor.Type == "e7" or
+			actor.Type == "gnrl" or
+			actor.Type == "e7.noautotarget" or
+			actor.Type == "gnrl.noautotarget"
+	end)
+
+	Trigger.OnAnyKilled(VIPs, function()
```

Comment:
You shouldn't do this at every tick. Do the trigger setup once after all player actors are spawned.